lib/instrumentation/index.js

```diff
@@ -164,6 +167,13 @@ Instrumentation.prototype.start = function () {
     }
   }
 
+  const lambdaHandlerInfo = getLambdaHandler(process.env)
+  if (lambdaHandlerInfo) {
+    this.addPatch(
+      lambdaHandlerInfo.filePath,
+      require('./modules/_lambda-handler')
```

Comment:
```suggestion
    (...args) => {
      return require(`./modules/_lambda-handler.js`)(...args)
    }
```

This *could* use the lazy-requiring technique used in the constructor, which would avoid loading _lambda-handler.js in the (admittedly rare) case where (a) lambda handler info can be loaded, but (b) the Lambda Runtime never gets to loading the user's function file, e.g. for handler name errors or some other runtime error.

Reply:
@trentm I'm not quite sure I understand what you're proposing here. Could you show me some pseudo code of what you'd like to see in the constructor?

Reply:
The constructor already uses a lazy-requiring technique:

```js
    this.addPatch(mod, (...args) => {
      // Lazy require so that we don't have to use `require.resolve` which
      // would fail in combination with Webpack. For more info see:
      // https://github.com/elastic/apm-agent-nodejs/pull/957
      return require(`./modules/${pathName}.js`)(...args)
    })
```

That was initially added in #957 to avoid `require`ing files in "lib/instrumentation/modules/" unless they were required. The same thing could also be done in the `this.addPatch(...)` call that you've added here so that ".../modules/_lambda-handler.js" is only `require`d when it is actually necessary.

No need to bother, however. The only time there will be `lambdaHandlerInfo` and *not* a need to `require('./modules/_lambda-handler.js')` is when there is some early Lambda Runtime error.

Reply:
Ah, I see.  By "lazy loading" you meant the way that `addPatch` can accept _either_ the module to be patched, or a function that will be called in order to load the module to be patched (the later being lazy loading).

Seems simple enough.  Done in 9ffc95131bf33df3e49aa5b044148d632b8afcd8